import_data/tasks/osm_update.py

```diff
@@ -0,0 +1,203 @@
+    ])
+    if exec_command(args) != 0:
+        log_error("Failed to run command `{}`".format(args))
+        return False
+    return True
+
+
+def check_settings(settings, keys):
+    errors = 0
+    for key in keys:
+        if settings.get(key) is None:
+            log_error("Missing `{}` setting".format(key))
+            errors += 1
+    return errors == 0
+
+
+def osm_update(ctx, pg_connection, osm_update_working_dir, imposm_data_dir, imposm_config_dir, change_file):
+    settings = {
+        "pg_connection": pg_connection,
+        "osm_update_working_dir": osm_update_working_dir,
+        "imposm_data_dir": imposm_data_dir,
+        "imposm_config_dir": imposm_config_dir,
+        "change_file": change_file,
+    }
+
+    # Settings
+    settings["start"] = int(datetime.now().timestamp())
+    settings["exec_time"] = get_time_now()
+    # imposm
+    if settings.get("imposm_config_dir", "") == "":
+        # default value, can be set with the --config option
+        settings["imposm_config_dir"] = "/etc/imposm"
+    # base tiles
+    settings["base_imposm_config_filename"] = "config_base.json"
+    # poi tiles
+    settings["poi_imposm_config_filename"] = "config_poi.json"
+    # tilerator
+    settings["from_zoom"] = 11
+    settings["before_zoom"] = 15 # exclusive
+
+    if not check_settings(settings, ["osm_update_working_dir", "imposm_data_dir"]):
+        return False
+
+    log("new osm_update process started")
+    log("working into directory: {}".format(settings["osm_update_working_dir"]))
+
+    if settings.get("change_file") is None:
+        log_error("A change file is required as input.")
+        return False
+    if not path.isfile(settings["change_file"]):
+        log_error("Change file `{}` was not found.".format(settings["change_file"]))
+        return False
+
+    # Update db and tiles, only if changes file is not empty
+    if os.path.getsize(settings["change_file"]) != 0:
+        # Imposm update for both tiles sources
+        if (not run_imposm_update(settings, settings["base_imposm_config_filename"])
+                or not run_imposm_update(settings, settings["poi_imposm_config_filename"])):
+            return False
+
+        # We make the import here to prevent a circular dependency if put at the top.
+        from .tasks import reindex_poi_geometries
+        # Reindex geometries to avoid index bloat
+        reindex_poi_geometries(ctx)
+
+        # Create tiles jobs for both tiles sources
+        if (not create_tiles_jobs(settings, settings["base_imposm_config_filename"])
+                or not create_tiles_jobs(settings, settings["poi_imposm_config_filename"])):
+            return False
+
+    log("============")
+    log("current location: {}".format(os.getcwd()))
+    log("============")
+    elapsed = settings["start"] - int(datetime.now().timestamp())
+    log("osm_update duration: {}h{:02}m{:02}s".format(
+        elapsed / 3600,
+        elapsed % 3600 / 60,
+        elapsed % 60))
```

Comment:
Some minor issues here with the duration format:
```suggestion
    elapsed = int(datetime.now().timestamp()) - settings["start"]
    log("osm_update duration: {}h{:02}m{:02}s".format(
        elapsed // 3600,
        elapsed % 3600 // 60,
        elapsed % 60))
```